### PR TITLE
Inspection Docs

### PR DESCRIPTION
**Pull request by i-walker**
This also adds features from the #114 PR and for enhanced Type search.

### CODE REVIEW

File: idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/inspection/InspectionSyntax.kt [hunk 1]
```diff
@@ -20,23 +20,80 @@ import com.intellij.openapi.project.Project
...
+   *     },
+   *     level = HighlightDisplayLevel.ERROR,
+   *     groupPath = arrayOf("Meta", "SimplePlugin")
+   *    )
+   *   )
+   *  }
+   * //sampleEnd
+   * ```
+   * Needless to say, the latter implementation is not sufficient enough as a purityPlugin, as the function body of the underlying Call's may have impure Call's.
+   * This Plugin will be discoverable in the user setting's under the Path Meta and it's displayName is `SimplePlugin`.
```
Comment (i-walker) [inline idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/inspection/InspectionSyntax.kt]:
```suggestion
   * This Plugin will be discoverable in the user setting's under the Path Meta and its groupDisplayName is `SimplePlugin`.
```

File: idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/inspection/InspectionSyntax.kt [hunk 2]
```diff
@@ -20,23 +20,80 @@ import com.intellij.openapi.project.Project
...
-import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtPsiFactory
 
 /**
- * More General Inspections can be build with [AbstractKotlinInspection] e.g.: [org.jetbrains.kotlin.idea.inspections.RedundantSuspendModifierInspection]
+ * Loosely speaking, `Inspection's` are easily recognized as "QuickFixes" when the user hit's a KeyShortCut for missing imports.
+ * Interestingly enough, despite calling `Inspection's` proverbially "QuickFixes", IntelliJ defines a `QuickFix` as an aggregation of multiple `Intention's`.
+ * Whereas `Intention's` analysis your code and users can decide whether they want to apply a suggested Fix,
+ * Inspection's improve upon that very idea and are capable to block the user to compile code at the first place.
+ * Additionally, we can scope the Fix we define in `applyTo` locally, for each instance per file, or globally to the whole project, assuming it has an universal refactoring task.
```
Comment (i-walker) [inline idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/inspection/InspectionSyntax.kt]:
```suggestion
 * Additionally, we can scope the Fix in `applyTo` locally, for each instance per file, or globally to the whole project, assuming it has a universal refactoring task.
```

File: .github/workflows/build_documentation.yml [hunk 3]
```diff
@@ -14,15 +14,9 @@ jobs:
...
       run: |
         set -ex
-        ./gradlew dokka
-    - name: Run Ank
-      env:
-        JAVA_OPTS: -Xms512m -Xmx1024m
-      run: |
-        set -ex
-        ./gradlew :docs:runAnk
+        ./gradlew dokka :docs:runAnk
```
[the remark this answers is not in the document]
Comment (i-walker) [reply .github/workflows/build_documentation.yml]: yes, we had issues in #264 and I am applying it to all upcoming docs as long #264 is not merged.
@rachelcarmena may be the one to ask about performance.

[the remark this answers is not in the document]
Comment (rachelcarmena) [reply .github/workflows/build_documentation.yml]: Hi! I think Gradle doesn't run the tasks in a linear way for all the modules. We joined them in the same line in order to manage dependencies properly.

**Pull request merged**